security/src/main/java/org/kaazing/gateway/security/auth/BasicLoginModule.java

```diff
@@ -61,16 +61,25 @@ public void initialize(Subject subject, CallbackHandler callbackHandler,
     protected boolean doLogin() throws LoginException {
 
         if (!authenticationSchemeIsBasic()) {
-            return true;
+            throw new LoginException("Only able to handle Basic authentication scheme");
```

Comment:
I see, after reading down below, I believe the multiple realm case, the first realm/chain for login modules is bypassed on the way back in (correct?), so then it won't get invoked.  So yes I believe throwing the LoginModule Exception here is the correct behavior